src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -272,17 +276,17 @@ internal static bool IsPathRelative(string path)
                 return true;
             }
 
-            if ((path[0] == '\\') || (path[0] == '/'))
+            if ((path[0] == Path.DirectorySeparatorChar) || (path[0] == Path.AltDirectorySeparatorChar))
             {
                 // There is no valid way to specify a relative path with two initial slashes
-                return !((path[1] == '\\') || (path[1] == '/'));
+                return !((path[1] == Path.DirectorySeparatorChar) || (path[1] == Path.AltDirectorySeparatorChar));
```

Comment:
We have an internal IsDirectorySeparator function somewhere.  Maybe it's on Path rather than on PathInternal?  Consider adding one to PathInternal (or moving it to PathInternal) to simplify these checks, e.g.
```C#
if (IsDirectorySeparator(path[0]))
{
    return !IsDirectorySeparator(path[1]);
}
```